packages/table/test/utilsTests.ts

```diff
@@ -303,4 +302,132 @@ describe("Utils", () => {
+            }
+        });
+
+        describe("with `keys` not defined", () => {
+            describe("returns true if values are shallowly equal", () => {
+                runTest(true, { a: 1, b: "2", c: true}, { a: 1, b: "2", c: true});
+                runTest(true, {}, {});
+                runTest(true, null, null);
+                runTest(true, undefined, undefined);
+            });
+
+            describe("returns false if values are not shallowly equal", () => {
+                runTest(false, undefined, null);
+                runTest(false, null, {});
+                runTest(false, {}, []);
+                runTest(false, { a: [1, "2", true] }, { a: [1, "2", true] });
+                runTest(false, { a: 1, b: "2", c: {}}, { a: 1, b: "2", c: {}});
+                runTest(false, { a: 1, b: "2", c: {}}, { a: 1, b: "2", c: {}});
+                runTest(false, { a: 1, b: "2", c: { a: 1 }}, { a: 1, b: "2", c: { a: 1 }});
+            });
+
+            describe("returns false if keys are different", () => {
+                runTest(false, {}, { a: 1 });
+                runTest(false, { a: 1 }, {});
+                runTest(false, { a: 1, b: "2" }, { b: "2" });
+                runTest(false, { b: "2" }, { a: 1, b: "2" });
+                runTest(false, { a: 1, b: "2", c: true}, { b: "2", c: true, d: 3 });
+            });
+
+            describe("returns true if same deeply-comparable instance is reused in both objects", () => {
+                const deeplyComparableThing1 = { a: 1 };
+                const deeplyComparableThing2 = [1, "2", true];
+                runTest(true, { a: 1, b: deeplyComparableThing1 }, { a: 1, b: deeplyComparableThing1 });
+                runTest(true, { a: 1, b: deeplyComparableThing2 }, { a: 1, b: deeplyComparableThing2 });
+            });
+
+            function runTest(expectedResult: boolean, a: any, b: any) {
+                it(`${JSON.stringify(a)} and ${JSON.stringify(b)}`, () => {
+                    expect(Utils.shallowCompareKeys(a, b)).to.equal(expectedResult);
+                });
+            }
+        });
+    });
+
+    describe("arraysEqual", () => {
+        describe("no compare function provided", () => {
+            describe("should return true if the arrays are shallowly equal", () => {
+                runTest(true, undefined, undefined);
+                runTest(true, null, null);
+                runTest(true, [], []);
+                runTest(true, [3], [3]);
+                runTest(true, [3, "1", true], [3, "1", true]);
+            });
+
+            describe("should return false if the arrays are not shallowly equal", () => {
+                runTest(false, undefined, null);
+                runTest(false, null, []);
+                runTest(false, null, [3]);
+                runTest(false, [3], []);
+                runTest(false, [3, 1, 2], [3, 1]);
+                runTest(false, [{}], [{}]);
+                runTest(false, [{ x: 1 }], [{ x: 1 }]);
```

Comment:
changes in this file generally feel like _too much testing_. most of these cases are equivalent: they're different instances of the same thing so of course they're not shallow equal. `[]` is the same as `[3]`, from my perspective.

i would do this with like 1-3 cases in each `describe` but not worry about testing every possible permutation.